packages/Multiselect/src/MultiSelectBar.tsx

```diff
@@ -26,6 +28,11 @@ const MultiSelectBar = ({
 }: any) => {
   const requiredClassName: string = buildClassNames({ required });
 
+  const hasNumber = typeof checkedPreviewCount === 'number' && isFinite(checkedPreviewCount);
+  const previewCount = hasNumber ? Math.max(0, Number(checkedPreviewCount)) : checkedItems.length;
+  const visibleItems = checkedItems.slice(0, previewCount);
```

Comment:
check in testing if this should be in a useMemo so it does make this for each render